starcheck/src/lib/Ska/Starcheck/Obsid.pm

```diff
@@ -23,14 +23,97 @@ use warnings;
 
 use Inline Python => q{
 import numpy as np
+from astropy.table import Table
+
+from mica.archive import aca_dark
 from chandra_aca.star_probs import guide_count
+from chandra_aca.transform import yagzag_to_pixels, count_rate_to_mag, mag_to_count_rate
 import Quaternion
 from Ska.quatutil import radec2yagzag
 import agasc
 
+from proseco.core import ACABox
+from proseco.guide import get_imposter_mags
+
 def _guide_count(mags, t_ccd):
     return float(guide_count(np.array(mags), t_ccd))
 
+def check_hot_pix(idxs, yags, zags, mags, types, t_ccd, date, dither_y, dither_z):
+    """
+    Return a list of info to make warnings on guide stars or fid lights with local dark map that gives an
+    'imposter_mag' that could perturb a centroid.  The potential worst-case offsets (ignoring effects
+    at the background pixels) are returned and checking against offset limits needs to be done
+    from calling code.
+
+    This fetches the dark current before the date of the observation and passes it to
+    proseco.get_imposter_mags with the star candidate positions to fetch the brightest
+    2x2 for each and calculates the mag for that region.  The worse case offset is then
+    added to an entry for the star index.
+
+    :param idxs: catalog indexes as list or array
+    :param yags: catalog yangs as list or array
+    :param zags: catalog zangs as list or array
+    :param mags: catalog mags (AGASC mags for stars estimated fid mags for fids) list or array
+    :param types: catalog TYPE (ACQ|BOT|FID|MON|GUI) as list or array
+    :param t_ccd: observation t_ccd in deg C (should be max t_ccd in guide phase)
+    :param date: observation date (bytestring via Inline)
+    :param dither_y: dither_y in arcsecs (guide dither)
+    :param dither_z: dither_z in arcsecs (guide dither)
+    :param yellow_lim: yellow limit centroid offset threshold limit (in arcsecs)
+    :param red_lim: red limit centroid offset threshold limit (in arcsecs)
+    :return imposters: list of dictionaries with keys that define the index, the imposter mag,
+             a 'status' key that has value 0 if the code to get the imposter mag ran successfully,
+             calculated centroid offset, and star or fid info to make a warning.
+    """
+
+    idxs = [int(idx) for idx in idxs]
```

Comment:
I've forgotten why all this input casting is required.  I understand the unicode thing, and casting from np array to pure Python types on output, but why the `float` / `int` casting?

Reply:
> on input I think the weak Perl types have caught me up

That's a fair enough answer (i.e. Perl idiotically passing `"1.2"` instead of a float).  Every time I see `$value + 0.0` it makes my stomach turn but that's what it is.

Reply:
I cut them anyway in a new commit at the end of this PR and did enough of the regression tests that I think this is fine.